docs/IE_PLUGIN_DG/NewTransformation.md

```diff
@@ -1,15 +1,390 @@
+
+```cpp
+// Register matcher and callback
+this->add_matcher(m, callback, PassProperty::CHANGE_DYNAMIC_STATE);
+```
+
+Also you can have multiple matchers and callbacks and they can be registered in single Graphrewrite pass. In this case all registered patterns will be applied in a singe graph traversal. 
+
+```cpp
+// Multiple matchers example
+this->add_matcher(m1, callback1, PassProperty::CHANGE_DYNAMIC_STATE);
+this->add_matcher(m2, callback2, PassProperty::CHANGE_DYNAMIC_STATE);
+```
+
+The last argument `PassProperty::CHANGE_DYNAMIC_STATE` says that callback can be applied for ngraph::Function with dynamic shapes. In case if callback does not support dynamic shapes `PassProperty::REQUIRE_STATIC_SHAPE` can be used.
+> **Note**: property mechanism will be deprecated soon and PassProperty::CHANGE_DYNAMIC_STATE is suggested to be used by default.
+
+To run any transformation you need to call `un_on_function(f)` method where `f` is `ngraph::Function`.
+```cpp
+ngraph::pass::MyTransformationClass().run_on_function(f);
+``` 
+  
+## Pattern matching <a name="pattern_matching"></a>
+
+Sometimes patterns can't be expressed via regular nGraph operations. For example if you want to detect Convolution->Add sub-graph without specifying particular input type for Convolution operation or you want to create pattern where some of operations can have different types.
+And for these cases nGraph provides additional helpers to construct patterns for GraphRewrite transformations. 
+There are two main helpers:
+1. `ngraph::pattern::op::Label` - helps to express inputs if their type is undefined.
+2. `ngraph::pattern::op::Any` - helps to express intermediate nodes of pattern if their type is unknown.
+
+Let's go through example to have better understanding how it works:
+
+> **Note**: node attributes do not participate in pattern matching and needed only for operations creation. Only operation types participate in pattern matching.
+
+Example below shows basic usage of `pattern::op::Label` class.
+Here we construct Multiply pattern with arbitrary first input and Constant as a second input.
+
+@snippet example_ngraph_utils.cpp pattern:label_example
+
+This example show how we can construct pattern when operation has arbitrary number of inputs.
+
+@snippet example_ngraph_utils.cpp pattern:concat_example
+
+This example shows how to use predicate to construct pattern where operation has two different types.
+
+@snippet example_ngraph_utils.cpp pattern:predicate_example
+
+TODO: add examples for ngraph::pattern::op::Any
+
+## Working with ngraph::Function <a name="working_with_ngraph_function"></a>
+
+In this chapter we will review nGraph API that allows us to manipulate with `ngraph::Function`.
+
+###1. ngraph::Node input and output ports
+
+First of all let's talk about `ngraph::Node` input/output ports. Each nGraph operation has input and output ports except cases when operation has Result, Parameter or Constant type.
```

Comment:
let's enclose `Result, Parameter or Constant` to ``

Reply:
done.